orchestrator/ModelScanner/Tasks/HashTask.cs

```diff
@@ -50,13 +50,33 @@ string ComputeCRC32Hash(Stream fileStream)
             return GetHashString(crc32.GetCurrentHash());
         }
 
+        // helper method to compute hash of the safetensor model without the header
+        string? ComputeAutoV3Hash(Stream fileStream)
+        {
+          var fileExtension = Path.GetExtension(filePath);
+
+          if (fileExtension != ".safetensors") {
+            return null;
+          }
+
+          fileStream.Position = 0;
+          var buffer = new byte[8];
+          fileStream.ReadExactly(buffer);
+          var metadataLength = BitConverter.ToInt64(buffer);
+          var metadataBuffer = new byte[metadataLength];
```

Comment:
Technically there is no need to allocate a buffer here as we're not going to read the header, we're just going to skip over